src/polycap-source.c

```diff
@@ -107,12 +108,28 @@ polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_rng *r
 	polycap_norm(&start_direction);
 
 	// Provide random electric vector
+	// 	make sure electric vector is perpendicular to photon direction
 	r = polycap_rng_uniform(rng);
-	start_electric_vector.x = (1.-2.*fabs(r));
-	r = polycap_rng_uniform(rng);
-	start_electric_vector.y = (1.-2.*fabs(r));
-	r = polycap_rng_uniform(rng);
-	start_electric_vector.z = (1.-2.*fabs(r));
+	if(fabs(r) <= source->hor_pol){
+		//horizontally polarised
+		start_electric_vector.x = 1.;
+		start_electric_vector.y = 0.;
+		start_electric_vector.z = 0.;
+	} else {
+		//vertically polarised
+		start_electric_vector.x = 0.;
+		start_electric_vector.y = 1.;
+		start_electric_vector.z = 0.;
+	}
+	cosalpha = polycap_scalar(start_electric_vector, start_direction);
+	alpha = acos(cosalpha);
+	c_ae = 1./sin(alpha);
+	c_be = -1.*c_ae*cosalpha;
+
+	start_electric_vector.x = start_electric_vector.x * c_ae + start_direction.x * c_be;
+	start_electric_vector.y = start_electric_vector.y * c_ae + start_direction.y * c_be;
+	start_electric_vector.z = start_electric_vector.z * c_ae + start_direction.z * c_be;
```

Comment:
momenteel de initiële electric vector als dusdanig bepaald, gebasseerd op `xmi_main.F90`. Ksnap nog steeds niet het gedoe met `c_ae` en `c_be`, maar denk dat het 'gewoon' geavanceerde vector projecties zijn (iets wat mocht ik het zelf moeten bedenken ik wss in 17 stappen ofzo op dezelfde uitkomst zou komen).

Enfin, mocht je het toch eens willen nakijken of het juist lijkt: graag.

Ik bemerkte trouwens ook dat mijn X en Y richting blijkbaar omgekeerd gedefinieerd zijn dan in xmimsim (althans bij mij is X de horizontale as, ik meen uit die polarisatie op te maken dat het bij xmimsim omgekeerd is?). Iets dat we in het achterhoofd moeten houden bij implementatie in XMIMSIM (of dat ik alsnog moet aanpassen, maar is een hoop werk...)

Reply:
Ik had me inderdaad gebasseerd op die geometry help in XMI-MSIM om er zeker van te zijn dat jouw assenstelsel anders is georiënteerd. Vraag is dus voornamelijk of dit volgens jou super problematisch is, of we er gewoon mee kunnen werken?
In principe kunnen we er mee werken als we maar voorzichtig genoeg te werk gaan bij het doorgeven van de X en Y coordinaten e.d.

Reply:
Zolang je goed oplet mag er in principe geen probleem zijn. Ben wel ietwat verbaasd dat Laszlo indertijd twee verschillende coordinaten systemen gebruikte voor zijn MC en polycap software...